Server/src/models/DataUploadModel.ts

```diff
@@ -15,190 +15,185 @@ import { Material } from './entities/Material';
 import { Datapointcomments } from './entities/Datapointcomments';
 import { Representations } from './entities/Representations';
 
+// Left logs in the class for debugging after DB migration.
 
-const insertReferenceType = async (preferenceType) => {
+/**
+ * This model class is responsible for updating the database with the extracted from the fileUpload. 
+ */
+export class DataUploadModel {
+    constructor() {
+    }
 
-    const connection = getConnection();
+    async insertReferenceType(preferenceType: string) {
 
-    let book = new Publicationtype();
-    book.id;
-    book.name = preferenceType;
-    await connection.manager.save(book);
-    return book.id;
-}
+        const connection = getConnection();
+
+        let book = new Publicationtype();
+        book.id;
+        book.name = preferenceType;
+        await connection.manager.save(book);
+        return book.id;
+    }
 
-const insertPublisher = async (publisher) => {
+    async insertPublisher(publisher: string) {
 
-    const connection = getConnection();
+        const connection = getConnection();
 
-    let publisherName = new Publisher();
-    publisherName.id;
-    publisherName.name = publisher;
-    await connection.manager.save(publisherName);
-    return publisherName.id;
-}
+        let publisherName = new Publisher();
+        publisherName.id;
```

Comment:
publisherName.id, to be removed?

Reply:
No its set that way inside its entity. Not needed to be specified

Reply:
No the entity has a primaryColumnKey that is autoincremented so we dont need to specify

Reply:
The value needs to be declared like this for the id to be correctly incremented in TypeORM. The same thing applies for any null value or value with a default entry.

Reply:
understood, resolved, thanks for the comments